Makefile

```diff
@@ -2,21 +2,21 @@ LARGE_TEST_REPO_DIR:=tests/test_docs_large
 GIT_DESC=$$(git describe)
 JUNIT_FLAGS := --capture=sys -o junit_logging=all
 
-.PHONY: lint-black
-lint-black: ## check python source code formatting issues, with black
-	black --check --diff ./
+.PHONY: lint-ruff
+lint-ruff: ## check the python source code with various linter rules through ruff
+	ruff check
 
-.PHONY: lint-black-apply
-lint-black-apply: ## apply black's source code formatting suggestions
-	black ./
+.PHONY: lint-ruff-apply
+lint-ruff-apply: ## apply all fixes made by ruff to the source code 
+	ruff check --fix
 
-.PHONY: lint-isort
-lint-isort: ## check imports are organized, with isort
-	isort --check --diff ./
+.PHONY: format-ruff
+format-ruff: ## Check the formatting of the python source code with ruff
+	ruff format --check
 
-.PHONY: lint-isort-apply
-lint-isort-apply: ## apply isort's imports organization suggestions
-	isort ./
+.PHONY: format-ruff-apply
```

Comment:
I added the phony targets since the previous black and isort targets had them, but the logic is to use `fix` and `check/lint` commands anyway so I've removed the extra phony targets.